run_core_ops.sh

```diff
@@ -0,0 +1,289 @@
+
+# Create output directories
+mkdir -p "${OUTPUT_DIR}"
+mkdir -p "${ORGANIZED_DIR}"
+
+# Create a comma-separated list of the 77 core ops
+CORE_OPS="abs,_adaptive_avg_pool2d,_adaptive_avg_pool2d_backward,add,addmm,any,avg_pool2d,avg_pool2d_backward,bitwise_and,bitwise_not,bitwise_xor,bmm,cat,clamp,clone,col2im,constant_pad_nd,convolution,convolution_backward,cos,cumsum,div,elu,eq,erf,exp,flip,floor,fmod,ge,gelu,grid_sampler_2d,gt,hardtanh,isinf,isnan,le,leaky_relu,log2,_log_softmax,lt,max,maximum,max_pool2d_with_indices,max_pool2d_with_indices_backward,mean,min,minimum,mm,mul,native_group_norm,native_group_norm_backward,native_layer_norm,ne,neg,nonzero,pow,reciprocal,reflection_pad2d,relu,remainder,repeat,round,rsqrt,sigmoid,sin,_softmax,split_with_sizes,sqrt,sub,sum,tanh,_to_copy,topk,upsample_bilinear2d,upsample_nearest2d,where"
+
+echo "Running KernelAgent on 77 core TorchBench operators..."
+echo "Output directory: ${OUTPUT_DIR}"
+echo "This will take a while as it generates and tests kernels for each operation."
+echo ""
+
+# Activate conda environment if needed
+if [ -n "$CONDA_PREFIX" ]; then
+    PYTHON_CMD="python"
+else
+    PYTHON_CMD="/home/leyuan/miniconda3/envs/backendbench/bin/python"
+fi
+
+# Set Python path
+export PYTHONPATH="/home/leyuan/workplace/BackendBench:$PYTHONPATH"
+
+# Run BackendBench with KernelAgent and capture output
+echo "Starting kernel generation at $(date)" | tee "${LOG_FILE}"
+$PYTHON_CMD BackendBench/scripts/main.py \
+    --suite torchbench \
+    --backend kernel_agent \
+    --ops "$CORE_OPS" \
+    --kernel-agent-workers 4 \
+    --kernel-agent-max-rounds 10 2>&1 | tee -a "${LOG_FILE}"
+
+echo ""
+echo "Kernel generation completed at $(date)" | tee -a "${LOG_FILE}"
+
+# Extract the generated kernels directory from the log
+KERNEL_RUN_DIR=$(grep -o "generated_kernels/kernel_agent_run_[0-9_]*" "${LOG_FILE}" | tail -1)
+
+if [ -z "$KERNEL_RUN_DIR" ] || [ ! -d "$KERNEL_RUN_DIR" ]; then
+    echo "ERROR: Could not find generated kernels directory"
+    exit 1
+fi
+
+echo "Found kernels in: $KERNEL_RUN_DIR"
+
+# Parse results and organize kernels
+echo "Organizing successful kernels..."
+
+# Create summary report
+cat > "${SUMMARY_FILE}" << EOF
+# KernelAgent Core Ops Run Summary
+**Date**: $(date)
+**Total Operations**: 77
+**Configuration**:
+- Workers: 4
+- Max Rounds: 10
+
+## Results
+
+| Operation | Status | Correctness | Performance | Location |
+|-----------|--------|-------------|-------------|----------|
+EOF
+
+# Create a detailed failure log
+FAILURE_LOG="${OUTPUT_DIR}/failures.md"
+cat > "${FAILURE_LOG}" << EOF
+# Failed Operations Debug Log
+**Date**: $(date)
+
+This log contains detailed information about operations that failed during kernel generation or BackendBench correctness checks.
+
+## Failed Operations
+
+EOF
+
+# Parse the log file for results and organize kernels
+$PYTHON_CMD << 'PYTHON_SCRIPT' "${LOG_FILE}" "${KERNEL_RUN_DIR}" "${ORGANIZED_DIR}" "${SUMMARY_FILE}" "${FAILURE_LOG}"
```

Comment:
Why is this logic in a .sh script. I feel like you'd want this functionality baked into backendbench generally?

https://github.com/meta-pytorch/BackendBench/pull/92 got merged in recently, and you should change the `save_verbose_results` to get the info you want (also feel free to add more info to the dict).